### ISSUE TITLE
Redo file if file size is not in desired range

### ISSUE THREAD

**Issue by fromage9747** [HaveAGitGat/Tdarr#568]
**Is your feature request related to a problem? Please describe.**
This is not a solution to a problem with the Tdarr application.  merely an enhancement to assist in the automation of the encode.

There are on occasion encodes that will either not be within the desired file size range or they even increase in file size after being encoded from x264 to x265.

**Describe the solution you'd like**
I would like Tdarr to be able to have a defined principle file size, and if it doesn't match that file size then to match desired file size ranges.  If the file size is too small, then to run an increased quality preset and if the file is too large, to run a decreased file size preset in order to obtain the desired file size/quality. 

As an example, there was a series that was a blu-ray, and every time I would encode it with my usual settings, it would be well under 1gb and the quality would be very blocky and awful.  So I would encode it again with an increased quality setting until I got it to 1gb <= 1.5gb in file size.  Which gave it the desired file size that I am able to accept as well as good quality. This of course I did manually through handbrake. 

Currently, I have a video series that is +-3gb on average in x264 and after it's encoded to 265 it's either the same size, a tiny little bit smaller or larger.  This is when I would like to be able to have Tdarr reencode the file with a preset I have defined when the file size is too large in order to get it to the desired size.

**Describe alternatives you've considered**
Currently, I have Tdarr set to delete the file after it is encoded, which is an acceptable risk for me with regard to series.  For movies, I move them to an x265_ripped folder to check before I manually delete them myself in order to ensure that the file sizes and quality is acceptable. 

I would most likely need to do the same thing for series as a workaround, but as there are many episodes, well... I would just like to be able to do the above.

**Additional context**
In Summary:
1. Tdarr to check file size after completing the encode
2. If file size is within a desired file size range, move on to the next one
3. If file size is too small, encode again with a predefined preset that has a higher quality setting
4. If the file size is too large, encode again with a predefined preset that has a lower quality setting.
5. Rinse and repeat.
6. There can be a failure threshold as well.  If Tdarr has encoded 3 times and the file size does still not meet the desired range. Move the file to failed encode and send the user an email notifying them of the Encode log and end file sizes etc.

**Comment by HaveAGitGat** [comment]
I think the ideal solution to this problem would be to set bitrate limits in the ffmpeg command so that the output file is guaranteed to be within your size limits so you don't have to waste time/energy re-transcoding.

**Comment by fromage9747** [comment]
Dam, thank you for your advice.  Wish I had found this out sooner. I had been using the Constant Quality option all this time in Handbrake.  Cheers

**Comment by HaveAGitGat** [comment]
No problem 👍
